### PR TITLE
04 get/api/reviews/id/comments

### PR DESCRIPTION
**Pull request by tuttitaken01**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -97,11 +97,77 @@ describe("2.getReviews", () => {
...
+                expect(comments).toHaveLength(3);
+            })
+        })
+        test("returns an empty array if id is valid but no comments are found", () => {
+            return request(app)
+            .get("/api/reviews/1/comments")
+            .expect(200)
+            .then(res => {
+                comments = res.body.comments;
+                expect(comments).toHaveLength(0);
```
Comment (aloush-dev) [inline __tests__/app.test.js:157]: This isn't specifically testing if its and array that comes back

File: __tests__/app.test.js [hunk 2]
```diff
@@ -97,11 +97,77 @@ describe("2.getReviews", () => {
...
+            })
+        })
+        test("returns a 404 error if input is not found", () => {
             return request(app)
-            .get("/api/review")
+            .get("/api/reviews/47/comments")
             .expect(404)
```
Comment (aloush-dev) [inline __tests__/app.test.js:163]: You should also test the message that comes back, and if there isn't one you should reject with a message aswell as a status code.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -97,11 +97,77 @@ describe("2.getReviews", () => {
...
         })
+        test("returns a 400 error if input is not number type", () => {
+            return request(app)
+            .get("/api/reviews/dj/comments")
+            .expect(400)
```
Comment (aloush-dev) [inline __tests__/app.test.js:168]: same as previous comment

File: models.js [hunk 4]
```diff
@@ -21,17 +21,32 @@ exports.selReviews = () => {
...
     GROUP BY reviews.review_id;`, [id])
     .then(result => {
-        console.log(result);
+        if(result.rows.length === 0) {
+            return Promise.reject({
+                status: 404,
+                message: "Not Found"
+            })
+        }
+        //console.log(result.rows[0]);
```
Comment (aloush-dev) [inline models.js:39]: remember to remove console logs

File: models.js [hunk 5]
```diff
@@ -21,17 +21,32 @@ exports.selReviews = () => {
...
+        return result.rows[0];
+    })
+} 
+
+exports.fetchComments = (id) => {
+    return db.query(`SELECT *
+    FROM comments
+    WHERE review_id=$1;`, [id])
+    .then(result => {
+        //console.log(result.rows);
```
Comment (aloush-dev) [inline models.js:49]: console logs

**Review by aloush-dev — APPROVED** [review]

File: __tests__/app.test.js (hunk 1 above)
Comment (tuttitaken01) [reply __tests__/app.test.js:157]: Not sure I fully understand this comment, sorry

**Pull request merged**